staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -156,35 +156,35 @@ func (p *Patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 		if patch == nil {
 			lookupPatchMeta, err = strategicpatch.NewPatchMetaFromStruct(versionedObject)
 			if err != nil {
-				return nil, nil, cmdutil.AddSourceToErr(fmt.Sprintf(createPatchErrFormat, original, modified, current), source, err)
+				return nil, cmdutil.AddSourceToErr(fmt.Sprintf(createPatchErrFormat, original, modified, current), source, err)
 			}
 			patch, err = strategicpatch.CreateThreeWayMergePatch(original, modified, current, lookupPatchMeta, p.Overwrite)
 			if err != nil {
-				return nil, nil, cmdutil.AddSourceToErr(fmt.Sprintf(createPatchErrFormat, original, modified, current), source, err)
+				return nil, cmdutil.AddSourceToErr(fmt.Sprintf(createPatchErrFormat, original, modified, current), source, err)
 			}
 		}
 	}
 
 	if string(patch) == "{}" {
-		return patch, obj, nil
+		return obj, nil
```

Comment:
We are returning same object and we can use reflect.DeepEqual safely.

Reply:
This is the perfect spot to say that we changed or not the resource with the patch. See my other comments about returning that information, as requested by Antoine.